### ISSUE TITLE
Reconcile CompareInfo.Unix code duplication because of Span extension methods

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/corefx#30678]
With https://github.com/dotnet/coreclr/pull/16467 we added methods for various Span extension methods that we added to MemoryExtensions(.Fast).cs. These methods are mostly just code duplicates of their string counterparts which call into an Interop function that accepts a pointer instead of a string. We should try to reconcile the diverged code paths to avoid duplication of huge code chunks. Described in this comment: https://github.com/dotnet/coreclr/pull/16467/files#r169709853

cc @ahsonkhan @tarekgh

**Comment by tarekgh** [comment]
I think @ahsonkhan looked at that before when implementing the Span APIs but I don't remember the reason why we didn't do that in the first place.

**Comment by ViktorHofer** [comment]
right, I talked with Ahson offline and that's a leftover from his work, that's why I opened the issue :)
